README.md

```diff
@@ -1,19 +1,97 @@
-# Homebrew Test Bot
+Homebrew Test Bot
+=================
 
-Tests the full lifecycle of a Homebrew change to a tap (Git repository).
+Homebrew comes with a comprehensive suite of tools that lets you
+validate that both your homebrew installation is working and prove that
+a formula is working.
 
-## Install
+Homebrew Test Bot uses these tools to simplify the automation of your
+formula publishing process including Checking your installation,
+Checking your Formula, and Building and uploading binaries.
+
+Install
+-------
 
 `brew test-bot` is automatically installed when first run.
 
-## Usage
+Usage
+-----
+
+Here are some of the most common use-cases for `brew test-bot`
+
+### Locally Validating a formula
+
+If this is for a custom tap, first, tap the repository.
+
+    brew tap "custom/tap"
+
+Then make your changes.
+
+    echo $YOUR_CHANGES > "$(brew --repo)/Library/Taps/custom/homebrew-tap/Formula/your-formula.rb"
+
+Next, we can validate our changes, including testing and linting, and
+dependency checks.
+
+    brew test-bot "custom/tap/your-formula"
+
+### Validating a formula change in GitHub actions
+
+You are running on GitHub actions test-bot will try to fix any potential
+problems in the homebrew repository deleting the `$(brew --repo)/Cellar`
+before performing any operations. If you've intentionally made changes
+there, they will be lost, what's more `--cleanup` is always passed.
+
+When `--cleanup` is passed, we hard reset the taps to "origin/master",
```

Comment:
I'd rather we point to (and update, if necessary) `--help` instead of documenting behaviour here that may change in future.